app/models/draft_vaccination_record.rb

```diff
@@ -154,6 +151,31 @@ def patient=(value)
 
   delegate :location, to: :session, allow_nil: true
 
+  def earliest_possible_value
+    session.academic_year.to_academic_year_date_range.first.beginning_of_day
+  end
+
+  def latest_possible_value
+    [
+      session.academic_year.to_academic_year_date_range.last.end_of_day,
+      Time.current
+    ].min
+  end
+
+  def performed_at_within_range
+    if performed_at < earliest_possible_value
+      errors.add(
+        :performed_at,
+        "Vaccination cannot take place before #{earliest_possible_value.to_fs(:long)}"
+      )
+    elsif performed_at > latest_possible_value
+      errors.add(
+        :performed_at,
+        "Vaccination cannot take place after #{latest_possible_value.to_fs(:long)}"
+      )
+    end
+  end
```

Comment:
I would suggest moving these methods in to the `private` section below as they shouldn't be called outside this class.